nurse_scheduling.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2020 D-Wave Systems Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This code includes an implementation of the algorithm described in Ikeda, 
+# K., # Nakamura, Y. & Humble, T.S. Application of Quantum Annealing to Nurse 
+# Scheduling Problem. Sci Rep 9, 12837 (2019). 
+# https://doi.org/10.1038/s41598-019-49172-3, © The Author(s) 2019, use of 
+# which is licensed under a Creative Commons Attribution 4.0 International 
+# License (To view a copy of this license, visit 
+# http://creativecommons.org/licenses/by/4.0/).
+
+from dwave.system import LeapHybridSampler
+from dimod import BinaryQuadraticModel
+
+# count nurses n = 1 ... N_NURSES
+# count scheduling days as d = 1 ... N_DAYS
+# binary variable q_nd is the assignment of nurse n to day d
+a = 3.5
+N_NURSES = 3
+N_DAYS = 11
+SIZE = N_DAYS * N_NURSES
+
+# Hard shift constraint: at least one nurse is working each day
+# implemented by penalizing any nurses scheduled on two successive
+# days
+
+# Hard nurse constraint: no nurse works two or more consecutive days
+# Lagrange parameter, for hard nurse constraint, on workforce and effort
+# Workforce function W(d) - set to a constant for now
+# Effort function E(n) - set to a constant for now
+lamda = 1.3
```

Comment:
lambda

Reply:
@JoelPasvolsky, if I use "lambda", it collides with the Python keyword. I'm open to other suggestions.

Reply:
`lagrange_multiplier`? A bird whispered in my ear

Reply:
OK good. I used it. Thanks.